### ISSUE TITLE
compilation error

### ISSUE THREAD

**Issue by oltolm** [cnjinhao/nana#682]
I get this compile error with f49f931be49c4d6890fbd2d56f7402017a9fb793

```
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/programming_interface.cpp: In function 'bool nana::api::keyboard_qwerty(nana::window, std::vector<std::__cxx11::basic_string<char> >, nana::keyboard_behaves, nana::keyboard_modes)':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/programming_interface.cpp:1703:24: error: 'fales' was not declared in this scope
[build]  1703 |                 return fales;
[build]       |                        ^~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/programming_interface.cpp: In function 'bool nana::api::keyboard_numeric(nana::window)':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/programming_interface.cpp:1714:24: error: 'fales' was not declared in this scope
[build]  1714 |                 return fales;
[build]       |                        ^~~~~
```

and this warning

```
[build] [29/85] Building CXX object _deps/nana-build/CMakeFiles/nana.dir/source/gui/dragdrop.cpp.obj
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/dragdrop.cpp: In static member function 'static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = void; _Functor = nana::dragdrop::dragdrop(nana::window)::<lambda(const nana::arg_mouse&)>; _ArgTypes = {const nana::arg_mouse&}]':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/dragdrop.cpp:1105:36: warning: 'executed_action' may be used uninitialized [-Wmaybe-uninitialized]
[build]  1105 |                         dnd_action executed_action;
[build]       |                                    ^~~~~~~~~~~~~~~
```

and more warnings

```
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp: In constructor 'nana::detail::keyboards::numeric::numeric(nana::window, nana::window, const nana::detail::keyboards::images*)':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:226:56: warning: 'nana::detail::keyboards::numeric::images_' will be initialized after [-Wreorder]
[build]   226 |                         const keyboards::images* const images_;
[build]       |                                                        ^~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:224:38: warning:   'nana::detail::basic_window* const nana::detail::keyboards::numeric::window_' [-Wreorder]
[build]   224 |                         window const window_;
[build]       |                                      ^~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:65:25: warning:   when initialized here [-Wreorder]
[build]    65 |                         numeric(window wd, window host, const keyboards::images* images ):
[build]       |                         ^~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp: In member function 'nana::rectangle nana::detail::keyboards::qwerty::_m_make_metrics()':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:597:37: warning: unused variable 'topline_keys' [-Wunused-variable]
[build]   597 |                                 int topline_keys = 12;
[build]       |                                     ^~~~~~~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp: In member function 'void nana::detail::keyboards::qwerty::_m_assign_key_values()':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:699:59: warning: comparison of integer expressions of different signedness: 'int' and 'std::__cxx11::basic_string<wchar_t>::size_type' {aka 'long long unsigned int'} [-Wsign-compare]
[build]   699 |                                         for (int i = 0; i < letters.size(); ++i)
[build]       |                                                         ~~^~~~~~~~~~~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:715:59: warning: comparison of integer expressions of different signedness: 'int' and 'std::__cxx11::basic_string<wchar_t>::size_type' {aka 'long long unsigned int'} [-Wsign-compare]
[build]   715 |                                         for (int i = 0; i < sym.size(); ++i)
[build]       |                                                         ~~^~~~~~~~~~~~
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp: In member function 'void nana::detail::keyboards::qwerty::_m_render_candidates(nana::paint::graphics&)':
[build] C:/src/CPM/nana/733c6e38b28c79e46d4d5b8c4a21bdeab53a98aa/source/gui/detail/virtual_keyboard.cpp:931:45: warning: unused variable 'cd_idx' [-Wunused-variable]
[build]   931 |                                 std::size_t cd_idx = 0;
[build]       |                                             ^~~~~~
```

**Comment by cnjinhao** [comment]
It a simple typo error, thank you.

**Comment by oltolm** [comment]
Thanks, now I get another error

```
[build] FAILED: _deps/nana-build/CMakeFiles/nana.dir/source/paint/image.cpp.obj 
[build] ccache C:\msys64\mingw64\bin\c++.exe -DNANA_ENABLE_VIRTUAL_KEYBOARD -DWIN32 -IC:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/include -O3 -march=native -mtune=native -DNDEBUG -Wall -fmax-errors=3 -MD -MT _deps/nana-build/CMakeFiles/nana.dir/source/paint/image.cpp.obj -MF _deps\nana-build\CMakeFiles\nana.dir\source\paint\image.cpp.obj.d -o _deps/nana-build/CMakeFiles/nana.dir/source/paint/image.cpp.obj -c C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/paint/image.cpp
[build] C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/paint/image.cpp: In static member function 'static HICON__* nana::paint::image_accessor::icon(const nana::paint::image&)':
[build] C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/paint/image.cpp:66:50: error: 'floor' is not a member of 'std'; did you mean 'std::chrono::floor'?
[build]    66 |                         auto row = (size_t) std::floor((i / 4) / size.height);
[build]       |                                                  ^~~~~
[build] In file included from C:/msys64/mingw64/include/c++/13.2.0/bits/fs_fwd.h:37,
[build]                  from C:/msys64/mingw64/include/c++/13.2.0/filesystem:48,
[build]                  from C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/include/nana/paint/graphics.hpp:20,
[build]                  from C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/include/nana/paint/image.hpp:17,
[build]                  from C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/detail/mswin/platform_spec.hpp:21,
[build]                  from C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/detail/platform_spec_selector.hpp:18,
[build]                  from C:/src/CPM/nana/8e8e86305b21f6b7260c45d46cc0f6a6f1a8279e/source/paint/image.cpp:15:
[build] C:/msys64/mingw64/include/c++/13.2.0/bits/chrono.h:1052:7: note: 'std::chrono::floor' declared here
[build]  1052 |       floor(const time_point<_Clock, _Dur>& __tp)
[build]       |       ^~~~~
```

**Comment by oltolm** [comment]
Thank you.
